src/sortStudents.ts

```diff
@@ -1,16 +1,90 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number [];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' |'desc';
 
+export function sortStudents(
+  students: Student [], sortBy: SortType, order: SortOrder,
+): Student [] {
+  const copyStudents: Student[] = students.map((obj:Student) => {
+    const tempStor = { ...obj };
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+    tempStor.grades = [...obj.grades];
+
+    return tempStor;
+  });
+
+  copyStudents.sort((a: Student, b: Student): number => {
+    const {
+      name: nameA,
+      surname: surnameA,
+      age: ageA,
+      married: marriedA,
+      grades: gradesA,
+    } = a;
+
+    const {
+      name: nameB,
+      surname: surnameB,
+      age: ageB,
+      married: marriedB,
+      grades: gradesB,
+    } = b;
+
+    const averageGradeA = gradesA.reduce((sum, n) => sum + n)
+      / a.grades.length;
+    const averageGradeB = gradesB.reduce((sum, n) => sum + n)
+    / b.grades.length;
+
+    const orderSwitcher = order === 'asc' ? 1 : -1;
+
+    switch (sortBy) {
+      case SortType.Name:
+        return nameA < nameB ? (-1 * orderSwitcher) : (1 * orderSwitcher);
```

Comment:
what are this 1/-1 are used for? why not use just `orderSwitcher`
```suggestion
        return nameA < nameB ? (-1 * orderSwitcher) : (1 * orderSwitcher);
```